prow/plank/controller.go

```diff
@@ -598,4 +607,49 @@ func (c *Controller) patchProwjob(srcPJ prowapi.ProwJob, destPJ prowapi.ProwJob)
 	_, err = c.prowJobClient.Patch(srcPJ.Name, ktypes.MergePatchType, patch)
 	c.log.WithFields(pjutil.ProwJobFields(&destPJ)).Debug("Patched ProwJob.")
 	return err
+
+}
+
+// Get the jobs that we need to retain so horologium can continue working
+// as intended.
+// And clean up old pods.
+func (c *Controller) cleanUpPods(pj prowapi.ProwJob) {
+
+	isFinished := sets.NewString()
+	isExist := sets.NewString()
+
+	// Now clean up old pods.
+	selector := fmt.Sprintf("%s = %s", kube.CreatedByProw, "true")
+	for _, client := range c.podClients {
+		pods, err := client.List(metav1.ListOptions{LabelSelector: selector})
+		if err != nil {
+			c.log.WithError(err).Error("Error listing pods.")
+			return
+		}
+		maxPodAge := c.config().Sinker.MaxPodAge.Duration
```

Comment:
Your test is failing here, because the `newFakeConfigAgent` is missing Sinker config...

Reply:
You should fix it in `controller_test.go` at the beginning of the file.